src/test/ui/consts/miri_unleashed/const_refers_to_static.rs

```diff
@@ -0,0 +1,38 @@
+// compile-flags: -Zunleash-the-miri-inside-of-you
+#![allow(const_err)]
+
+#![feature(const_raw_ptr_deref)]
+
+use std::sync::atomic::AtomicUsize;
+use std::sync::atomic::Ordering;
+
+const BOO: &usize = { //~ ERROR undefined behavior to use this value
+    static FOO: AtomicUsize = AtomicUsize::new(0);
+    unsafe { &*(&FOO as *const _ as *const usize) }
+    //~^ WARN skipping const checks
+};
+
+const FOO: usize = {
+    static FOO: AtomicUsize = AtomicUsize::new(0);
+    FOO.fetch_add(1, Ordering::Relaxed) // FIXME: this should error
```

Comment:
Wait how can this not error...?

Reply:
I'm also baffled.

* `fetch_add` calls https://github.com/rust-lang/rust/blob/c605199e89572e586a5f37bc698c48b6a10896fb/src/libcore/sync/atomic.rs#L1595
* `atomic_add` calls https://github.com/rust-lang/rust/blob/c605199e89572e586a5f37bc698c48b6a10896fb/src/libcore/sync/atomic.rs#L2262
* that intrinsic *does not exist* in the const eval machine

oh hahah

```rust
#![allow(const_err)]
```

at the top of this file silences the error, and since `const FOO` is not actually used anywhere, no error.

Reply:
I'll make it `warn`

Reply:
I'd rather you added some actual uses, and allowed everything also in the other tests. Having unallowable hard errors for this feels better. ;)

Reply:
The actual usese won't give you any helpful errors. It will just give you "use of erroneous constant"

Reply:
Oh right, diagnostics with `allow(const_err)` are really bad then. We had an open issue about this somewhere, right?

Reply:
yes: https://github.com/rust-lang/rust/issues/65597